iohub/dataset_base.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from pathlib import Path
+from types import TracebackType
+from typing import Any, Iterable, Optional, Type, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+_AXES_PREFIX = ["T", "C", "Z", "Y", "X"]
+
+
+class BaseFOV(ABC):
+    _root: Path
+
+    @property
+    @abstractmethod
+    def axes_names(self) -> list[str]:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def channels(self) -> list[str]:
```

Comment:
Is there a "correct" way to enforce that `len(BaseFOV.channels)` should be identical to `BaseFOV.shape[1]` within the abstract base class?

Similarly, if we include a `BaseFOV.times` then `len(BaseFOV.times)` should be identical to `BaseFOV.shape[0]`.

Reply:
I'm not really sure what's the best practice. I think that should be covered by tests, another option is to add an assert in this class but it will only be useful if `super().channels` is called.